web-console/knikubevirt/vm-details/vm-details.md

```diff
@@ -0,0 +1,114 @@
+# VM Details
+
+
+
+## Overview Tab
+
+![VM-overview](img/1-0-0.jpg)
+The user clicks a VM’s name from the list view and is taken to that VM’s Overview tab.
+The user sees detailed information about the VM’s configuration.
+
+
+## YAML
+
+Should be inline with the rest of Openshift. Can be seen [here](http://openshift.github.io/openshift-origin-design/web-console/future-openshift/code-editor-updates/code-editor-updates)
+
+
+## Network Interfaces
+
+![VM-network intefaces - add NIC](img/2-0-0.jpg)
+This is Network Interfaces tab.
+
+
+![VM - add NIC modal](img/2-0-1.jpg)
+Default modal. The user is notified that activating this upcoming NIC requires a VM restart. 
+
+![VM - add NIC modal - filled in](img/2-0-2.jpg)
+User can choose between adding NIC only, or add NIC and restart VM to make it active.
+
+![VM - new NIC](img/2-0-3.jpg)
+The new NIC should always appear at the top of the list until the user restarts it. The new nic will have a'Pending VM restart' Link state (as a status).
+
+![VM - New NIC - popover](img/2-0-4.jpg)
+Hover on that link state (status) will display more details with a 'Restart VM action in a popover.
+
+![VM - Pending changes notification - view changes](img/2-0-5.jpg)
+Whenever a NIC has been added or modified in a way that requires the VM to be restarted, an inline notification should appear below the tab area reminding the user to do so. This notification should persist across all tab views, including the Overview (shown previously).
+
+![VM - Pending changes notification - view changes modal](img/2-0-6.jpg)
+Changes pending VM restart can be viewed in a dedicated modal.
+
+### Edit NIC
+
+![VM - NIC item action options](img/2-1-0.jpg)
+
+![VM - Edit NIC modal](img/2-1-1.jpg)
+
+![VM - Edit NIC saved changes](img/2-1-2.jpg)
+
+
+## Disks
+
+![VM - Disks tab - adding disk](img/3-0-0.jpg)
+The flow for disks is similar to NIC.
+however, adding or editing a disk do not require VM restart.
+
+![VM - add disk modal](img/3-1-0.jpg)
+The user clicks “Add Disk”, and is lead to an 'Add disk' modal.
+
+![VM - disk list item actions](img/3-2-0.jpg)
+The user edits each input and clicks the confirm button to finish. The new disk is added to the list and is sorted alphabetically.
+
+The user can edit or delete a list item from its kebab menu for each list item.
+
+
+
+## Snapshots
+
+Snapshot designs can be seen [here](https://github.com/openshift/openshift-origin-design/pull/183)
+
+
+## Events
+
+TBD
+
+## Consoles
+
+![VM - console tab -default](img/4-0-0.jpg)
+The Consoles tab allows the user to connect to the virtual machine via an in-browser or desktop-based Graphical and/or Serial console.
+![VM - console dropdown options](img/4-1-0.jpg)
+The user can change the console type using the dropdown selector. 'Graphical (VNC)' is selected by default. Any currently selected option is disabled in the dropdown. 'Graphical (RDP/VNC)' open in separate windows, designated by the fa-external-link icon.
```

Comment:
My knowledge about this is outdated, but in [Marek's video](https://youtu.be/mLeCZcSHROo?t=347) it looks like Graphical (VNC) no longer has to be a separate window. That would mean removing the external link icon here.

Reply:
Will go with sticking to the same window since it is the desired solution.